service/src/main/kotlin/fi/espoo/evaka/calendarevent/CalendarEventQueries.kt

```diff
@@ -358,13 +381,77 @@ WHERE cp.period && ce.period
+     child_placement AS NOT MATERIALIZED (SELECT p.id,
+                                                 p.unit_id,
+                                                 p.child_id,
+                                                 daterange(p.start_date, p.end_date, '[]') as period
+                                          FROM placement p
+                                          WHERE EXISTS(SELECT 1 FROM children_of_guardian WHERE p.child_id = children_of_guardian.id)
+                                            AND daterange(p.start_date, p.end_date, '[]') && ${bind(range)})
+SELECT ce.id,
+       cp.child_id,
+       ce.period * daterange(dgp.start_date, dgp.end_date, '[]') * cp.period as period,
+       ce.period                                                             as event_period,
+       ce.title,
+       ce.description,
+       (
+           CASE
+               WHEN cea.child_id IS NOT NULL THEN 'individual'
+               WHEN cea.group_id IS NOT NULL THEN 'group'
+               ELSE 'unit' END
+           )                                                                 as type,
+       dg.id                                                                 as group_id,
+       dg.name                                                               as group_name,
+       unit.id                                                               as unit_id,
+       unit.name                                                             as unit_name,
+       cet.id                                                                as event_time_id,
+       cet.date                                                              as event_time_date,
+       cet.start_time                                                        as event_time_start,
+       cet.end_time                                                          as event_time_end,
+       cet.child_id                                                          as event_time_occupant
+FROM child_placement cp
+         LEFT JOIN daycare_group_placement dgp ON dgp.daycare_placement_id = cp.id
+         LEFT JOIN calendar_event_attendee cea
+                   ON cea.unit_id = cp.unit_id
+                       AND (cea.child_id IS NULL OR cea.child_id = cp.child_id)
+                       AND (cea.group_id IS NULL OR cea.group_id = dgp.daycare_group_id)
+         JOIN calendar_event ce ON ce.id = cea.calendar_event_id
+         LEFT JOIN daycare_group dg ON dg.id = cea.group_id
+         LEFT JOIN daycare unit ON unit.id = cea.unit_id
+         JOIN calendar_event_time cet
+                  ON ce.id = cet.calendar_event_id
+                       AND (cet.child_id IS NULL OR cet.child_id = cp.child_id)
+WHERE cp.period && ce.period
+  AND ce.period && ${bind(range)}
+  AND daterange(dgp.start_date, dgp.end_date, '[]') && ce.period
+  AND daterange(dgp.start_date, dgp.end_date, '[]') && cp.period
+  AND ce.event_type = 'DISCUSSION_SURVEY';
+"""
```

Comment:
En tiedä kannattaako enää lähteä muuttamaan, mutta itse olisin varmaan lähtenyt toteuttamaan tätä niin että palautetaan yhdessä queryssä calendar eventit ja toisessa niihin liittyvät ajat.

Eka query olisi muotoa:
```
SELECT ...
FROM calendar_event ce
JOIN calendar_event_attendee cea ON ...
JOIN daycare unit ON ...
LEFT JOIN daycare_group dg ON ...
WHERE ...
```
Siinä filtteröinnin voi hoitaa samalla tavalla WITH-kyselyillä tai suoraan alikyselyinä.

Toinen query olisi paljon yksinkertaisempi:
```
SELECT ...
FROM calendar_event_time cet
WHERE cet.calendar_event_id = ANY(${bind(calendarEventIds)})
```
ja queryn tuloksille `.groupBy { it.calendarEventId }`